exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -218,13 +218,18 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
-   * If cluster version is the same as default, update option to current drillbit version.
+   * If cluster version is the same as default, updates option to current drillbit version.
+   * Since getOption and setOption are separate calls, raise condition might occur
+   * when several drillbits are registering at the same time.
+   * It is assumed that the first drillbit in cluster sets cluster version
+   * but when the raise condition occurs, it might be the second or the third etc.
+   * This behaviour does not impose significant impact, since the main goal is to detect mismatching versions.
    */
-  private void checkAndUpdateClusterVersionOption(OptionManager optionManager, DrillbitEndpoint drillbitEndpoint) {
-    OptionValue versionOption = optionManager.getOption(ExecConstants.CLUSTER_VERSION);
+  private void checkAndUpdateClusterVersionOption(OptionManager systemOptions, DrillbitEndpoint drillbitEndpoint) {
```

Comment:
@paul-rogers 
your alternative state-free solution totally makes sense. I'll make appropriate changes.
Thank you!